### PR TITLE
Expand diacritics

### PR DESCRIPTION
**Pull request by Captain-Quack**
I was torn on weather to include greek characters, decided to include them.

### CODE REVIEW

File: server/database.js [hunk 1]
```diff
@@ -42,15 +42,29 @@ function escapeRegExp(string) {
...
+        ['[xẋξ]'],
         ['[yÿ]'],
-        ['[sśš]'],
-        ['[zžźż]'],
-        ['[cçćč]'],
-        ['[nñń]'],
+        ['[zžźżʒǯζ]'],
     ];
 
     const allCharacters = new RegExp(characterGroups.map(group => group[0]).join('|'), 'gi');
```
Comment (Captain-Quack) [inline server/database.js:70]: This could be expanded upon later to include multi-character substitutions like "mu" or even "eszett"

**Review by g3ner1c — APPROVED** [review]
It might be worth considering adding multiletter transliterations in the future, such as:

German "ß" as "ss" or "sz"
Icelandic "þ" as "th"
Nordic "æ" as "ae"

**Comment by Captain-Quack** [comment]
something like `['((ae)|æ)']` could work

**Comment by geoffrey-wu** [comment]
Ran the backend tests; for searching cesaire 5 times, the old method took 16.285s, this method before any optimizations took 18.592s, and this method after optimizations took 15.355s. All of these times are well within error.

**Pull request merged**